lib/kube/proxy/portforward_spdy.go

```diff
@@ -174,77 +173,54 @@ func (h *portForwardProxy) forwardStreamPair(p *httpStreamPair, remotePort int64
 
 	// read and write from the error stream
 	targetErrorStream, err := h.targetConn.CreateStream(headers)
+	h.onPortForward(fmt.Sprintf("%v:%v", h.podName, remotePort), err == nil /* success */)
 	if err != nil {
-		h.onPortForward(fmt.Sprintf("%v:%v", h.podName, remotePort), false)
-		return trace.ConnectionProblem(err, "error creating error stream for port %d", remotePort)
+		err := trace.ConnectionProblem(err, "error creating error stream for port %d", remotePort)
+		p.sendErr(err)
+		p.close()
```

Comment:
I noticed we close `p` on errors, but not on success. Who "owns" `p`? Is it feasible to make a single place responsible for closing it?

An alternative is to document the behavior of `forwardStreamPair` in regards to `p`, so others looking at it knows what it's expected to do.

Reply:
Moved the closing logic to the `portForward` function.